en/reads-and-writes.html

```diff
@@ -214,39 +214,58 @@ <h2 id="api-and-utilities">API and utilities</h2>
+The following operations are rejected:
+<ul>
+  <li>Put operations</li>
+  <li>Update operations that are <em>not</em> <code><a href="reference/document-json-format.html#assign">assign</a></code></li>
+  <li>Update <code><a href="reference/document-json-format.html#assign">assign</a></code> operations to multi-value fields</li>
+  <li>Update <code><a href="reference/document-json-format.html#assign">assign</a></code> operations to non-numeric single-value fields</li>
+</ul>
+
+These operations are still allowed:
+<ul>
+  <li>Remove operations</li>
+  <li>Update <code><a href="reference/document-json-format.html#assign">assign</a></code> operations to numeric single-value fields</li>
+</ul>
+</p>
+<p>
 To remedy, add nodes to the content cluster, or use nodes with higher capacity.
-The data will <a href="elastic-vespa.html">auto-redistribute</a>, and feeding is unblocked.
+The data will <a href="elastic-vespa.html">auto-redistribute</a>, and feeding is unblocked when all content nodes are below the limits.
 Configure <a href="reference/services-content.html#resource-limits">resource-limits</a> to tune this.
-Temporarily increasing <em>resource-limits</em> (e.g. from 0.8 to 0.9) is an alternative if the cluster is stuck.
-</p><p>
-These <a href="reference/metrics.html">metrics</a> indicate whether feeding is blocked (set to 1 when blocked):
+</p>
+<p>
+These <a href="reference/metrics.html">metrics</a> are used to monitor resource usage and whether feeding is blocked:
 <table class="table">
   <tr>
-    <th>content.proton.resource_usage.feeding_blocked</th>
-    <td>disk or memory</td>
-  </tr><tr>
-    <th>content.proton.documentdb.attribute.resource_usage.feeding_blocked</th>
-    <td><em>Attribute Multivalue Mapping</em> or <em>Attribute Enum Store</em></td>
+    <th>cluster-controller.resource_usage.nodes_above_limit</th>
+    <td>The number of content nodes that are above one or more resource limits. When above 0, feeding is blocked.</td>
+  </tr>
+  <tr>
+    <th>content.proton.resource_usage.disk</th>
+    <td>A number between 0 and 1, indicating how huch disk (of total available) is used on the content node.</td>
+  </tr>
+  <tr>
+    <th>content.proton.resource_usage.memory</th>
+    <td>A number between 0 and 1, indicating how huch memory (of total available) is used on the content node.</td>
   </tr>
 </table>
-When feeding is blocked, events are logged - examples:
+When feeding is blocked, error messages are returned in write operation replies - example:
 <pre>
-java.lang.RuntimeException: ReturnCode(NO_SPACE, Put operation rejected for document 'id:test:test::0': 'diskLimitReached: {
-  action: \"add more content nodes\",
-  reason: \"disk used (0.85) &gt; disk limit (0.8)\",
-  capacity: 100000000000,
-  free: 85000000000,
-  available: 85000000000,
-  diskLimit: 0.8
-}'
+ReturnCode(NO_SPACE, External feed is blocked due to resource exhaustion:
+                     memory on node 0 [example.com] (0.82 > 0.80))
```

Comment:
Should hostname example be e.g. `my-vespa-node-0.example.com` to make it more obvious that it's the hostname of the actual node?

Reply:
Fixed